tests/gpu/device_func.cpp

```diff
@@ -0,0 +1,6 @@
+#include "gpu_common.h"
+#include <cstdio>
+
+__device__ void device_function(int a) { printf("device_func %d\n", a); }
+
+__global__ void kernel2() { printf("Kernel2\n"); }
```

Comment:
Either delete `kernel2` or create another test that has only device functions. 

The inclusion of a kernel, will enable your pass to perform additional actions (e.g. there is kernel stub and it will instrument it). But we need to test the case that there is no such stub.

Reply:
Can you also change:

`__device__ void device_function(int a) { printf("device_func %d\n", a); } ` 

to 

__device__ void device_function(int *a) {*a  = 123;}

Thus avoiding the TU to contain an external function?

Reply:
Why avoid the external function? I'm using it to check using lit